WordPress/Sniffs/Theme/DeregisterScriptSniff.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Sniff to check theme for wp_deregister_style()
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_DeregisterScriptSniff extends WordPress_Sniffs_Functions_FunctionRestrictionsSniff
+{
+
+	/**
+	 * Exclude groups
+	 *
+	 * Example: 'switch_to_blog,user_meta'
+	 *
+	 * @var string Comma-delimited group list
+	 */
+	public $exclude = '';
+
+	/**
+	 * Groups of functions to restrict
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type' => 'error' | 'warning',
+	 * 		'message' => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'wp_deregister_script' => array(
+				'type'	=> 'error',
```

Comment:
I feel the problem should be solved upstream with the popular plugin and not having the temp fix in the theme. The theme can always deuque the old script on certain pages and load their own version using a different handle.